### PR TITLE
doc(dialog-basics): add webpack notes

### PR DESCRIPTION
**Pull request by StrahilKazlachev**
Added some notes related to #298.
@jods4 could you check if there is something I need to correct/rephrase/add?

### CODE REVIEW

File: doc/article/en-US/dialog-basics.md [hunk 1]
```diff
@@ -237,10 +241,28 @@ You can specify global settings as well for all dialogs to use when installing t
...
+    +-- app.js
+    +-- forms
+        +-- consent-form.js
+        +-- consent-form.html
+    +-- prompts
+        +-- prompt.js
+        +-- prompt.html
+    ```
+    If you want to open a `prompt` from `consent-form` the path will be `prompts/prompt`.
+    > Warning: WebPack users should use `PLATFORM.moduleName` when passing a path. Not using it may lead to the resource not being bundled, when not referenced anywhere else.
```
Comment (jods4) [inline doc/article/en-US/dialog-basics.md]: "when not referenced anywhere else"
It's more subtle than that. 
`moduleName` also ensures other properties important for `aurelia-loader`, like the assignment of a module id based on path, preserving exports and preventing their renaming, ensuring that the module doesn't get merged by `ModuleConcatenationPlugin`...

Bottom line: if a string is a dependency loaded by `aurelia-loader`, _always_ mark it as such with `moduleName`.

File: doc/article/en-US/dialog-basics.md [hunk 2]
```diff
@@ -237,10 +241,28 @@ You can specify global settings as well for all dialogs to use when installing t
...
+  - object - it will be used as the view model. In this case `view` must also be specified.
+  - class - the view model *class* or *constructor function*.
+    > Note: This approach depends on being able resolve the origin of the class.
+
+    > Warning: Known WebPack issues when using *DllPlugin*, *ModuleConcatenationPlugin*.
```
Comment (jods4) [inline doc/article/en-US/dialog-basics.md]: I would rephrase: "this approach might not work when using `DLLPlugin` or `ModuleConcatenationPlugin`."
And add a note to the effect of:
"Webpack users are advised to always use the `string` overload wrapped with `moduleName`"

**Comment by jods4** [comment]
I proposed tiny precisions to your changed. Other than that, looked fine to me!

**Comment by StrahilKazlachev** [comment]
@jods4 I didn't see any info on `PLATFORM.moduleName` in the Webpack section of the hub and why it is **important**. So I hope you don't mind I took one of your comments as is.

**Comment by jods4** [comment]
The hub still describes the v1.x plugins.
In the meantime, there's a lot of info about the new v2.x plugins in `aurelia-webpack-plugin` wiki, though.

Webpack traces a precise graph of each module's dependencies. 
`moduleName` is important because it teaches webpack about `aurelia-loader` dynamic dependencies.

If your project is not with Webpack you don't have to use it.
If you build a library or share code with others, then you should. It doesn't hurt to have it in the code as `aurelia-pal` implementation is just a no-op.

**Comment by StrahilKazlachev** [comment]
@jods4 If you are OK with the current changes I'll merge it.
About `PLATFORM.moduleName`, I just wanted to point to more info about it(or add in the dialog docs, as I did), so that we're not just saying that it *must be used* but and *why*.

File: doc/article/en-US/dialog-basics.md [hunk 3]
```diff
@@ -75,18 +75,21 @@ typings install github:aurelia/dialog --save
...
-      .plugin('aurelia-dialog');
+      .plugin(PLATFORM.moduleName('aurelia-dialog'));
 
     aurelia.start().then(a => a.setRoot());
   }
   </source-code>
 </code-listing>
 
-> Note: If you are using WebPack it is possible that the plugin is installed before Aurelia has replaced the `<body>` element, if that is where your `aurelia-app="main"` is defined, which results in some of the dialog components getting overwritten.  In this case you can move the `aurelia-app` attribute to a `<div>` inside of the `<body>`.  Example - `<body><div aurelia-app="main"></div></body>`.
+> Note: If you are using Webpack it is possible that the plugin is installed before Aurelia has replaced the `<body>` element, if that is where your `aurelia-app="main"` is defined, which results in some of the dialog components getting overwritten.  In this case you can move the `aurelia-app` attribute to a `<div>` inside of the `<body>`.  Example - `<body><div aurelia-app="main"></div></body>`.
```
Comment (jods4) [inline doc/article/en-US/dialog-basics.md]: Reading this again and it looks nasty to me.
Isn't that a timing bug in `aurelia-dialog`?
If you put stuff inside the `<body>` couldn't you just wait on Aurelia's loaded event?

**Comment by jods4** [comment]
Looks ok to me, but I just noted that the first addition is quite nasty. 
It's the kind of things that should "just" work out of the box :(

As far as explaining _why_ the best I can offer is to point users to this documentation: https://github.com/aurelia/webpack-plugin/wiki

**Comment by StrahilKazlachev** [comment]
@jods4 You are right about the nasty part. The reason I mentioned the Hub was I wanted to link to it, not the wiki, but it'll be probably better.
Thanks for the help 😄

**Comment by StrahilKazlachev** [comment]
@jods4 So I tracked why that *nasty* part was added - https://github.com/aurelia/dialog/issues/233#issuecomment-260755619, I'll see how to *properly* address it.

**Comment by jods4** [comment]
If I might: I don't like advice without explanation as it tends to turn into cargo cult.

Instead of just saying that it might get dismissed, could you say under which conditions? "when your app is the <body> tag, any dialog opened by the first page loaded in the application before `attached` and not properly awaited will be dismissed"

**Comment by jods4** [comment]
'the body tag
